src/lib/kdf/kdf.h

```diff
@@ -62,14 +63,17 @@ class BOTAN_PUBLIC_API(2, 0) KDF {
       * @param label purpose for the derived keying material
       * @param label_len size of label in bytes
       */
-      virtual void kdf(uint8_t key[],
-                       size_t key_len,
-                       const uint8_t secret[],
-                       size_t secret_len,
-                       const uint8_t salt[],
-                       size_t salt_len,
-                       const uint8_t label[],
-                       size_t label_len) const = 0;
+      BOTAN_DEPRECATED("Use KDF::derive_key")
+      void kdf(uint8_t key[],
+               size_t key_len,
+               const uint8_t secret[],
+               size_t secret_len,
+               const uint8_t salt[],
+               size_t salt_len,
+               const uint8_t label[],
+               size_t label_len) const {
+         perform_kdf({key, key_len}, {secret, secret_len}, {salt, salt_len}, {label, label_len});
```

Comment:
nit - when deprecating a function because it's supplanted by something else, I like to have the inlined implementation act as documentation of sorts of exactly how one should translate the call. Here we deprecate it saying use `derive_key`, but the implementation calls the internal `perform_kdf` instead, which isn't something a user can immediately copy and paste.

Reply:
Sounds like a good convention to me, thanks. I'll keep that in mind.